components/collection/HeroButtons.vue

```diff
@@ -130,6 +134,48 @@ const QRModalActive = ref(false)
 
 const hashtags = 'KusamaNetwork,KodaDot'
 const sharingLabel = $i18n.t('sharing.collection')
+
+const { transaction } = useTransaction()
+const isLoading = ref(false)
+
+const checkCollection = async (id) => {
+  type Collection = {
+    collectionEntity?: {
+      burned: boolean
+    }
+  }
+
+  const { data }: { data: Ref<Collection> } = await useAsyncQuery({
+    query: collectionByIdMicro,
+    clientId: client.value,
+    variables: {
+      id: id.toString(),
+    },
+  })
+
+  if (!isLoading.value) {
+    return
+  }
+
+  if (data.value.collectionEntity?.burned) {
+    navigateTo(`/${urlPrefix.value}/u/${accountId.value}?tab=collections`)
+  } else {
+    await delay(DETAIL_TIMEOUT)
```

Comment:
can we avoid the delay?
maybe by watching something and triggering the checkCollection function at the right time?

Reply:
Hmm, any idea?

Because I can't rely on `blocknumber`. To make sure the collection was burned, then navigate to the user collection page

Reply:
`/composables/useSubscriptionGraphql`  maybe?

Reply:
checking

Reply:
Thanks, it looks better with subscriptions. I forgot that we had that composable